src/GitVersion.Core/VersionCalculation/SemanticVersioning/ReferenceNameExtensions.cs

```diff
@@ -0,0 +1,45 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Contracts;
+using System.Text.RegularExpressions;
+using GitVersion.Extensions;
+
+namespace GitVersion;
+
+public static class ReferenceNameExtensions
+{
+    public static bool TryGetSemanticVersion(this ReferenceName source,
```

Comment:
Is there a reason for not having the method `TryGetSemanticVersion` on the `ReferenceName` class directly?

Reply:
Actually I have added it befor and got an compile error in the solution new-cli/GitVersion.sln. The ReferenceName.cs file has been added just as a reference... The SemanticVersion type is not known in this solution. @arturcic What do you suggest?